plugin/rollback.go

```diff
@@ -219,59 +244,65 @@ func (b *backend) serviceAccountPolicyRollback(ctx context.Context, req *logical
 	return err
 }
 
-func (b *backend) deleteServiceAccount(ctx context.Context, iamAdmin *iam.Service, account gcputil.ServiceAccountId) error {
-	if account.EmailOrId == "" {
-		return nil
+func (b *backend) serviceAccountPolicyDiffRollback(ctx context.Context, req *logical.Request, data interface{}) error {
+	b.rolesetLock.Lock()
+	defer b.rolesetLock.Unlock()
+
+	var entry walIamPolicyStaticAccount
+	if err := mapstructure.Decode(data, &entry); err != nil {
+		return err
 	}
 
-	_, err := iamAdmin.Projects.ServiceAccounts.Delete(account.ResourceName()).Do()
+	var rolesInUse util.StringSet
+
+	// Try to verify service account not being used by roleset
+	sa, err := b.getStaticAccount(entry.StaticAccount, ctx, req.Storage)
 	if err != nil {
-		if !isGoogleAccountNotFoundErr(err) && !isGoogleAccountUnauthorizedErr(err) {
-			return errwrap.Wrapf("unable to delete service account: {{err}}", err)
-		}
+		return err
 	}
-	return nil
-}
-
-func (b *backend) deleteTokenGenKey(ctx context.Context, iamAdmin *iam.Service, tgen *TokenGenerator) error {
-	if tgen == nil || tgen.KeyName == "" {
-		return nil
+	if sa.ResourceName() == entry.AccountId.ResourceName() {
```

Comment:
We should add a nil check here on `sa` to avoid a panic.

Reply:
I've added a check and fixed the missing `StaticAccount` fields in the WAL entries.

Reply:
Ty!